assets/javascript/app.js

```diff
@@ -0,0 +1,235 @@
+        console.log("Errors handled: " + errorObject.code);
+      });
+
+      trainRef.on("child_changed", changeTrain, function(errorObject) {
+        console.log("Errors handled: " + errorObject.code);
+      });
+      // [END_EXCLUDE]
+    } else {
+      // User is signed out.
+      // [START_EXCLUDE]
+      $("#signInText").text('Sign In');
+      trainRef = database.ref("/trains");
+      switchTrains();
+      // [END_EXCLUDE]
+    }
+    // [START_EXCLUDE]
+    // [END_EXCLUDE]
+  });
+  // [END authstatelistener]
+
+	function switchTrains(){
+      $("#timeTrains").empty();
+      trainRef.off("child_added");
+      trainRef.off("child_changed");
+      trainRef.on("child_added", newTrain, function(errorObject) {
+        console.log("Errors handled: " + errorObject.code);
+      });
+
+      trainRef.on("child_changed", changeTrain, function(errorObject) {
+        console.log("Errors handled: " + errorObject.code);
+      });
+	}
+
+  $("#signIn").on("click", toggleSignIn);
+
+
+  // trainRef.on("child_added", newTrain, function(errorObject) {
+  //   console.log("Errors handled: " + errorObject.code);
+  // });
+
+  // trainRef.on("child_changed", changeTrain, function(errorObject) {
+  //   console.log("Errors handled: " + errorObject.code);
+  // });
+
+  $("#addTrain").on("submit", function getNewTrain(event) {
+    event.preventDefault();
+    trainRef.push({
+      name: event.target.trainName.value.trim(),
+      destination: event.target.trainDest.value.trim(),
+      first: event.target.trainTime.value,
+      frequency: parseInt(event.target.trainFreq.value)
+    });
+    event.target.reset();
+  });
+
+  $("#updateTrain").on("submit", function updateTrain(event) {
+    event.preventDefault();
+    var key = event.target.dataset.key;
+    trainRef.child(event.target.dataset.key).update({
+      name: event.target.trainName.value.trim(),
+      destination: event.target.trainDest.value.trim(),
+      first: event.target.trainTime.value,
+      frequency: parseInt(event.target.trainFreq.value)
+    });
+    $("#updateModal").modal("hide");
+  });
+
+  $(document).on("click", ".remove", function removeRow(event) {
+    event.preventDefault();
+    var result = confirm("Are you sure you want to delete this train?");
+    if (result) {
+      trainRef.child(this.dataset.key).remove();
+      $(this).parent().parent().remove();
+    }
+  });
+
+  $(document).on("click", ".update", function updateRow(event) {
+    event.preventDefault();
+    trainRef.child(this.dataset.key).once("value").then(function(snapshot) {
+      var data = snapshot.val();
+      var form = $("#updateTrain")[0];
+      form.trainName.value = data.name;
+      form.trainDest.value = data.destination;
+      form.trainTime.value = data.first;
+      form.trainFreq.value = data.frequency;
+      $("#updateTrain").attr("data-key", snapshot.key);
+      $("#updateModal").modal("show");
+    });
+  });
+
+  var update = setInterval(updateTimes, 60000);
```

Comment:
This isn't being passed around or modified in any way so it doesn't seem necessary to store this interval as a variable, so I would either not store it or have some function that gets called that has the ability to clear, restart, or adjust the interval cycle time.